.github/workflows/agent-validation.yml

```
@@ -0,0 +1,86 @@
+          echo "Changed files:\n$CHANGED"
+          
+          # Ensure no edits to collaboration/state/**
+          if echo "$CHANGED" | grep -E '^collaboration/state/' >/dev/null; then
+            echo "❌ Out-of-scope change: collaboration/state/** is not allowed in PRs" >&2
+            exit 1
+          fi
+          
+          # Extract task id from PR title or branch name
+          TASK_ID=$(printf "%s\n%s" "$PR_TITLE" "$HEAD_REF" | sed -n 's/.*\(T-[0-9][0-9][0-9]\).*/\1/p' | head -n1 || true)
+          echo "Detected TASK_ID: ${TASK_ID:-none}"
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)